src/main/java/org/osc/controller/nsfc/api/NeutronSfcSdnRedirectionApi.java

```diff
@@ -307,8 +374,55 @@ public void deleteNetworkElement(NetworkElement serviceFunctionChain) throws Exc
     @Override
     public List<NetworkElement> getNetworkElements(NetworkElement serviceFunctionChain) throws Exception {
 
+        // NEW //////////////////////////////////////////
         this.utils.throwExceptionIfNullElementAndId(serviceFunctionChain, "Service Function Chain Id");
 
+        PortChain portChain = this.osClient.sfc().portchains().get(serviceFunctionChain.getElementId());
+        this.utils.throwExceptionIfCannotFindById(portChain, "Service Function Chain", serviceFunctionChain.getElementId());
+
+        if (portChain.getPortPairGroups() == null) {
+            return emptyList();
+        }
+
+        ServiceFunctionChainEntity sfcFound = new ServiceFunctionChainEntity(portChain.getId());
```

Comment:
seems like we are mixing old/new code...would avoid that and just create the new objects/classes